java/operator/src/main/java/com/instaclustr/cassandra/operator/controller/DataCenterReconciliationController.java

```diff
@@ -284,11 +284,11 @@ private void createOrReplaceStateNodesStatefulSet(final Iterable<ConfigMapVolume
 
     private V1Container fileLimitInit() {
         return new V1Container()
-                .securityContext(new V1SecurityContext().privileged(true))
+                .securityContext(new V1SecurityContext().privileged(dataCenterSpec.getPrivilegedSupported()))
```

Comment:
At some point we should do better validation of the CRD object, since this could be null.

Reply:
I'm going to mark this as a required field in the CRD. It's not super defensive at the operator level, but at least it should catch the vast majority of NPE this could generate.